src/shogun/classifier/svm/NewtonSVM.cpp

```diff
@@ -287,8 +196,8 @@ void CNewtonSVM::line_search_linear(float64_t* weights, float64_t* d, float64_t*
 			Xsv[i]=Xd[sv[i]];
 		}
 
-		memset(temp1.vector, 0, sizeof(float64_t)*sv_len);
-		SGVector<float64_t>::vector_multiply(temp1.vector, outzsv.vector, Ysv.vector, sv_len);
+		temp1.zero();
+		SGVector<float64_t>::vector_multiply(temp1.data(), outzsv.data(), Ysv.data(), sv_len);
```

Comment:
i left it like this because using linalg here will lead to allocating 3 more temp vectors each iteration of this loop and then a copy to temp1

Reply:
it will still have to be linalg, as we are planning to remove the SGVector interface soon.
Creating SGVector instances on stack is essentially for free so no worries.
You can even do them inline
`linalg::dot(SGVector<...>(...., false), SGVector<...>(...., false))`
no need to give them a name